mlflow/models/evaluation/__init__.py

```diff
@@ -0,0 +1,399 @@
+        """
+        raise NotImplementedError()
+
+    def evaluate(
+            self,
+            model: PyFuncModel,
+            model_type,
+            dataset,
+            run_id=None,
+            evaluator_config=None,
+            **kwargs
+    ) -> EvaluationResult:
+        """
+        :param model: A pyfunc model instance.
+        :param model_type: A string describing the model type (e.g., "regressor",
+                   "classifier", …).
+        :param dataset: An instance of `EvaluationDataset` containing features
+                        and labels (optional) for model evaluation.
+        :param run_id: The ID of the MLflow Run to which to log results.
+        :param evaluator_config: A dictionary of additional configurations for
+                                 the evaluator.
+        :param **kwargs: For forwards compatibility, a placeholder for additional
+                         arguments that may be added to the evaluation interface
+                         in the future.
+        :return: An `EvaluationResult` instance containing evaluation results.
+        """
+        client = mlflow.tracking.MlflowClient()
+        self.mlflow_client = client
+
+        def do_evaluate(_run_id):
+            timestamp = int(time.time() * 1000)
+            existing_dataset_metadata_str = client.get_run(_run_id).data.tags.get("mlflow.datasets")
+            if existing_dataset_metadata_str is not None:
+                dataset_metadata_list = json.loads(existing_dataset_metadata_str)
+            else:
+                dataset_metadata_list = []
+
+            metadata_exists = False
+            for metadata in dataset_metadata_list:
+                if (
+                    metadata["hash"] == dataset.hash
+                    and metadata["name"] == dataset._user_specified_name
+                ):
+                    metadata_exists = True
+                    break
+
+            if not metadata_exists:
+                dataset_metadata_list.append(dataset._metadata)
+
+            dataset_metadata_str = json.dumps(dataset_metadata_list)
+
+            metrics_dict, artifacts_dict = self.compute_metrics_and_compute_and_log_artifacts(
+                model, model_type, dataset, evaluator_config, _run_id
+            )
+            client.log_batch(
+                _run_id,
+                metrics=[
+                    Metric(key=f"{key}_on_{dataset.name}", value=value, timestamp=timestamp, step=0)
+                    for key, value in metrics_dict.items()
+                ],
+                tags=[RunTag("mlflow.datasets", dataset_metadata_str)],
+            )
+
+            return EvaluationResult(metrics_dict, artifacts_dict)
+
+        if mlflow.active_run() is not None:
+            return do_evaluate(mlflow.active_run().info.run_id)
+        else:
+            with mlflow.start_run(run_id=run_id) as run:
+                return do_evaluate(run.info.run_id)
+
+
+def evaluate(
```

Comment:
Can we also introduce a `list_evaluators()` API to list all of the available evaluators?

Reply:
After we implement the default evaluator, can we also import this `evaluate` API in https://github.com/mlflow/mlflow/blob/master/mlflow/__init__.py and  https://github.com/mlflow/mlflow/blob/master/mlflow/models/__init__.py so that users can call this via `mlflow.evaluate()` and `mlflow.models.evaluate()`?